R/logger.R

```diff
@@ -293,8 +294,25 @@ log_level <- function(level, ..., namespace = NA_character_,
         do.call(log_fun, log_arg)
 
     }
+
+    class(log_arg) <- "logger"
```

Comment:
note to self:

- this only preserves the last def, so change the `for` loop for `lapply` before return
- update `logger` to return formatted messages as well
- record `definitions` here so that we don't need to look up later

I will make these updates and experiment a bit more hopefully later today